src/_pytest/pytester.py

```diff
@@ -544,7 +544,11 @@ def __init__(self, request: FixtureRequest, tmpdir_factory: TempdirFactory) -> N
         self._mod_collections = (
             WeakKeyDictionary()
         )  # type: WeakKeyDictionary[Module, List[Union[Item, Collector]]]
-        name = request.function.__name__
+        if request.function:
+            name = request.function.__name__  # type: str
+        else:
+            name = request.node.name
+        self._name = name
```

Comment:
@RonnyPfannschmidt 
Would like to get your view on this: is a fallback to `request.node.name` sane, or is there something better?